apps/src/applab/designElements/PropertyRow.jsx

```diff
@@ -61,7 +94,8 @@ var PropertyRow = React.createClass({
         type={this.props.isNumber ? 'number' : undefined}
         value={this.state.value}
         onChange={this.handleChangeInternal}
-        style={rowStyle.input} />;
+        onBlur={this.props.isIdRow ? this.onIdRowBlur : null}
```

Comment:
I thought this way led to slightly better separation of id-specific logic, but I agree it's pretty much a toss-up.